docs/Hardlinks/How-to-setup-for/TrueNAS-Core.md

```diff
@@ -0,0 +1,132 @@
+# TrueNAS Core
+
+!!! note
+
+    This guide is based on TrueNAS Core 13, a storage solution based on ZFS. While there are more similarities than differences between Core and Scale in terms of creating datasets, this guide explicitly covers dataset, share, and basic permission setup in order to provide a place for creating and accessing the recommended folder structure. This does not cover jails or other virtualization means on the TrueNAS Core platform.
+
+    This guide assumes that you've already created a pool, which by default will contain a top-level dataset that was configured during the time of pool configuration. For the examples below, we'll be using a pool name of 'tank'.
+
+    Additionally, since SMB does not support hardlinks we will only be covering the creation and use of NFS shares.
+
+------
+
+## Create the main dataset
+
+!!! warning
+
+    To get hardlinks and atomic moves working with ZFS you will need to make use of <u>**ONE**</u> dataset with subfolders.
+
+For this example I'm using my own setup and preferred share `Media`. The naming isn't important as host machine mappings can be manipulated as desired.
+
+Go to Storage -> Pools and click on ⋮ on the right side of your root dataset, then select `Add Dataset`.
+
+![truenas-dataset](images/truenas-create-dataset.png)
+
+On the next screen you'll fill out your basic dataset info, such as name and comments. I have a few things grayed out here as the dataset has already been configured, but in this example you'd use `Media` in the Name field and keep everything else at its default (`Inherit`). During share creation, there will be an option at the bottom for `Share Type`. `Generic` is the default, with `SMB` as a selectable option. Since we'll be using NFS, select `Generic`. Optionally, you can click `Advanced Options` and populate `Quota for this dataset` if you want to limit what the dataset displays and will accept when it's mounted. Click `Submit` (mine shows `Save` since this is an existing dataset) once complete.
+
```

Comment:
A couple of things:
- `recordsize` ZFS property. I have mine set to 1M
- `compression` should be set to `zstd` as it's faster and compresses better
- `atime` is best set to `off` as it's pretty useless on media

Wether you want these options set on this specific dataset, or on the pool, is up to you. Personally I have set both `compression` and `atime` properties on the pool and have them as `inherit` in the dataset. the `recordsize` I set per dataset as it's use case specific.

I dont know the details of TrueNAS, but I would also enable encryption.

Reply:
encryption on media is indeed ... only for the extra files (srt, nfo, etc)
I mentioned zstd as it is now the default in FreeBSD :)

But thank you, good job!

Reply:
@TRaSH- changes recorded. Hopefully this is the last revision and we can get an initial version up. We can always tweak it later.